pkg/interface/link/src/js/components/link.js

```diff
@@ -18,33 +18,35 @@ export class LinkDetail extends Component {
 
     this.setComment = this.setComment.bind(this);
   }
-  
+
   componentDidMount() {
-    // if we have preloaded our data,
-    // but no comments, grab the comments
-    if (!!this.props.data.url) {
-      let props = this.props;
-      let comments = !!props.data.comments;
-      
-      if (!comments) {
-        api.getComments(props.path, props.data.url, props.page, props.link);
+    // if we have no preloaded data, and we aren't expecting it, get it
+    if (this.props.page != 0 && (!this.props.data || !this.props.data.url)) {
+      api.getPage(this.props.path, this.props.page);
+    } else {
+
+      // if we have preloaded our data,
+      // but no comments, grab the comments
+      if (!!this.props.comments) {
```

Comment:
Going directly to a submission page fails because of this: it can't get length of undefined on api.getCommentsPage, so it's passing an undefined URL — so, we want to wait for the URL. Might just need the same check at L48 here.

Reply:
Yeah, this is one of those "on a plane" clunky things I think — Links in general needs a cleanup and refactor it didn't get after "minimum viable working thing". :/ We'll see what we can do.